### ISSUE TITLE
As a DevSecOps engineer, I want to implement a scheduled backup of Postgres to GCP Cloud Storage

### ISSUE THREAD

**Issue by adaniels-parabol** [ParabolInc/parabol#4885]
As a DevSecOps engineer, I want to implement a scheduled backup of Postgres to GCP Cloud Storage

**Notes**
* @tiffanyhan I added you as an owner to the GCP project where the Cloud Storage would be provisioned
* @tiffanyhan here is the project/branch in Gitlab for backups, you can work in here https://gitlab.com/parabol1/action-devsecops/digitalocean-postgres/-/tree/backups-and-restores we already have backups being exported as a .tar artifact on manual run
* @tiffanyhan you can use this code chunk to get your started https://gitlab.com/parabol1/action-devsecops/digitalocean-postgres/-/blob/backups-and-restores/.gitlab-ci.yml#L226-286

AC:

- [x] Bucket created in GCP with private permissions (not public)
- [x] Add required CI/CD vars for GCP keys to Gitlab (https://gitlab.com/parabol1/action-devsecops/digitalocean-postgres/-/settings/ci_cd) they should be masked
- [x] Create GCP backup stages that run automatically when dynamic variable GCP_BACKUP is true
- [x] GCP backup should backup all environments daily to GCP Bucket. You can setup a schedule here https://gitlab.com/parabol1/action-devsecops/digitalocean-postgres/-/pipeline_schedules this is where the GCP_BACKUP variable will be used
- [x] Expected outcome is to have scheduled job run with generated backups being stored in GCP storage for each environment, with each environment having it's own folder (development, staging, production). Backups should be date/time stamped.

**Comment by tiffanyhan** [comment]
Notes:

- I used AWS instead of GCP, as we already have assets and backups stored in AWS and I thought it would be easier to consolidate everything in one cloud provider.
- I found that it wasn't necessary to set a dynamic env var when scheduling the pipeline. I set it up by simply targeting the `backup-and-restores` branch, and using `rules` keyword on the backup jobs.
- I tweaked the `pg_dump` command so that we are also compressing the tarball before sending to s3.
- I tried to make the filename of postgres backups equivalent to filename format of rethink backups. this turned out to be hard bc vars can't be shared btwn jobs, and global vars can't be set from bash commands. solution is likely to write the filename to a file artifact, which can then be picked up and read by the next job. I decided the ROE wasn't worth it. i instead got as close as possible: `postgres_backup_${CI_COMMIT_TIMESTAMP}.tar.gz`.
